pyveda/vedaset/store/vedabase.py

```diff
@@ -214,10 +212,25 @@ def __del__(self):
 
 
 class VedaBase(H5DataBase):
-
     class _MetaSample(tables.IsDescription):
         vid = tables.StringCol(36)
 
+    def __init__(self, *args, dataset_id=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        if dataset_id is not None:
+            self._root._v_attrs["dataset_id"] = dataset_id
+
+    def __iter__(self):
+        img_iter = self._root.images.iterrows()
+        lbl_iter = self._root.labels.iterrows()
+        return(zip(img_iter, lbl_iter))
+
+    def __next__(self):
+       try:
+            return super().__next__()
```

Comment:
So if we return self in `__iter__`, but still do the array iterator initialization like you have up there ^, you could do some nice thing like just define the thing you want to return in `__next__`: one simple implementation might look like:
```
def __next__(self):
     img = next(self._root.images)
     lbl = next(self._root.lables)
     # Do we need to get an id from the table here? Can talk about this next
     return (img, lbl)
```

The nice thing about that pattern is that we don't have to raise a StopIteration ourselves, we just let the underlying iterators do it themselves.